tests/integration/ambient/istioctl_test.go

```diff
@@ -0,0 +1,183 @@
+				t.Fatalf("Failed to unmarshal zc all output: %v", err)
+			}
+
+			// get the config dump generated when running the istioctl zc svc command and unmarshal it into ZtunnelDump struct
+			// for test comparison
+			args = []string{
+				"--namespace=dummy",
+				"zc", "svc", podName, "-o", "json",
+			}
+			zcSvcOutput, _ := istioCtl.InvokeOrFail(t, args)
+			if err = jsonUnmarshalListOrMap([]byte(zcSvcOutput), &dumpParsed.Services); err != nil {
+				t.Fatalf("Failed to unmarshal zc svc output: %v", err)
+			}
+			// need to initialize the SubjectAltNames field to an empty slice to avoid nil / slice comparison that fails the test
+			for _, svc := range dumpParsed.Services {
+				if svc.SubjectAltNames == nil {
+					svc.SubjectAltNames = []string{}
+				}
+			}
+
+			// get the config dump generated when running the istioctl policies command and unmarshal it into ZtunnelDump struct
+			// for test comparison
+			args = []string{
+				"--namespace=dummy",
+				"zc", "policies", podName, "-o", "json",
+			}
+			zcPoliciesOutput, _ := istioCtl.InvokeOrFail(t, args)
+			if err = jsonUnmarshalListOrMap([]byte(zcPoliciesOutput), &dumpParsed.Policies); err != nil {
+				t.Fatalf("Failed to unmarshal zc policies output: %v", err)
+			}
+
+			// get the config dump generated when running the istioctl zc workloads command and unmarshal it into ZtunnelDump struct
+			// for test comparison
+			args = []string{
+				"--namespace=dummy",
+				"zc", "workloads", podName, "-o", "json",
+			}
+			zcWorkloadsOutput, _ := istioCtl.InvokeOrFail(t, args)
+			if err = jsonUnmarshalListOrMap([]byte(zcWorkloadsOutput), &dumpParsed.Workloads); err != nil {
+				t.Fatalf("Failed to unmarshal zc workloads output: %v", err)
+			}
+
+			// get the config dump generated when running the istioctl zc certs command and unmarshal it into ZtunnelDump struct
+			// for test comparison
+			args = []string{
+				"--namespace=dummy",
+				"zc", "certs", podName, "-o", "json",
+			}
+			zcCertsOutput, _ := istioCtl.InvokeOrFail(t, args)
+			if err = jsonUnmarshalListOrMap([]byte(zcCertsOutput), &dumpParsed.Certificates); err != nil {
+				t.Fatalf("Failed to unmarshal zc certs output: %v", err)
+			}
+
+			// sort the slices to avoid false negative comparison between the raw dump and the parsed dump
+			//
+			// sort certificates by identity
+			slices.SortStableFunc(dumpAll.Certificates, func(a, b *configdump.CertsDump) int {
+				return strings.Compare(a.Identity, b.Identity)
+			})
+			slices.SortStableFunc(dumpParsed.Certificates, func(a, b *configdump.CertsDump) int {
+				return strings.Compare(a.Identity, b.Identity)
+			})
+			// sort workloads by UID
+			slices.SortStableFunc(dumpAll.Workloads, func(a, b *configdump.ZtunnelWorkload) int {
+				return strings.Compare(a.UID, b.UID)
+			})
+			slices.SortStableFunc(dumpParsed.Workloads, func(a, b *configdump.ZtunnelWorkload) int {
+				return strings.Compare(a.UID, b.UID)
+			})
+			// sort services by VIP
```

Comment:
Services weren't being sorted correctly in the last PR because some hostnames were the same. VIP/addresses seem to be unique for the service (is this an ok assumption?) so I'm grabbing the first one and sorting on that.

Reply:
context: https://github.com/istio/istio/commit/844592bac170a6c0384452e6e91801bf9aed3703